client/transaction-pool/tests/pool.rs

```diff
@@ -560,9 +589,11 @@ fn fork_aware_finalization() {
 
 	let e1;
 
-	// block e1
+	// block E1
 	{
-		let header = pool.api().push_block(5, vec![from_dave, from_bob], true);
+		// let header = pool.api().push_block(5, vec![from_dave, from_bob], true);
+		let header = pool.api().push_block_with_parent(d1, vec![from_dave, from_bob], true);
```

Comment:
DQ: Why do you changed this?

Reply:
I explained it in: https://github.com/paritytech/substrate/pull/12305/files#r982204082

Please cross-check if my reasoning is correct.

Reply:
Speculating - this maybe related to the `isbestblock` _bug_ in the `testapi` implementation.